pilot/pkg/serviceregistry/kube/controller/ambient/sidecar_interop.go

```diff
@@ -33,12 +33,31 @@ import (
 // This is to map to the eventual EDS structure.
 type serviceEDS struct {
 	ServiceKey       string
-	WaypointInstance []*workloadapi.Workload
+	WaypointInstance []model.WorkloadInfo
 	UseWaypoint      bool
 }
 
-func (w serviceEDS) ResourceName() string {
-	return w.ServiceKey
+func (s serviceEDS) ResourceName() string {
+	return s.ServiceKey
+}
+
+func (s serviceEDS) Equals(other serviceEDS) bool {
+	if s.ServiceKey != other.ServiceKey {
+		return false
+	}
+	if s.UseWaypoint != other.UseWaypoint {
+		return false
+	}
+	if len(s.WaypointInstance) != len(other.WaypointInstance) {
+		return false
+	}
+	// assumes builder sorted the slices
+	for i := range s.WaypointInstance {
+		if s.WaypointInstance[i].Workload.Uid != other.WaypointInstance[i].Workload.Uid {
```

Comment:
Is just checking the UID sufficient? this won't handle the workload changing things like labels, ip,etc?

Reply:
ah, this was calling Equals before I squashed. thanks for catching it